collectors/nvd/collectors.py

```diff
@@ -295,6 +324,11 @@ def update_cvss_via_flawcvss(self, flaw: Flaw, item: dict) -> bool:
                     continue
 
                 if original_cvss != new_cvss:
+                    # set impact if it was empty because a validation requires it
+                    if not flaw.impact:
+                        flaw.impact = self.calculate_highest_impact(
+                            [cvss2_score, cvss3_score, cvss4_score]
+                        )
```

Comment:
I think that the procedure should probably be a little bit different favoring the newer score versions before the old ones as they might have better expressing capabilities and therefore more precise results but I might be wrong and the caution is OK as a first estimate. Definitely leave it as it is. We can reconsider if someone complains.

Reply:
`[cvss2_score, cvss3_score, cvss4_score]` is created from new scores, e.g. `new_cvss4, cvss4_score = self.get_new_nvd_cvss(item, FlawCVSS.CVSSVersion.VERSION4)`